src/code/FindHelper.cs

```diff
@@ -128,177 +142,716 @@ public FindHelper(CancellationToken cancellationToken, PSCmdlet cmdletPassedIn)
-                    if (_type == ResourceType.None)
-                    {
-                        _cmdletPassedIn.WriteVerbose("Null Type provided, so add PSGalleryScripts repository");
-                        repositoriesToSearch.Insert(i + 1, psGalleryScripts);
-                    }
-                    else if (_type != ResourceType.None && _type == ResourceType.Script)
+                    foreach (PSResourceInfo currentPkg in SearchFromLocalRepository(repositoriesToSearch[i]))
                     {
-                        _cmdletPassedIn.WriteVerbose("Type Script provided, so add PSGalleryScripts and remove PSGallery (Modules only) from search consideration");
-                        repositoriesToSearch.Insert(i + 1, psGalleryScripts);
-                        repositoriesToSearch.RemoveAt(i); // remove PSGallery
+                        yield return currentPkg;
                     }
                 }
-                else if (String.Equals(repositoriesToSearch[i].Uri.AbsoluteUri, _poshTestGalleryUri, StringComparison.InvariantCultureIgnoreCase))
+                else
                 {
-                    // special case: for PoshTestGallery, Module and Script resources have different endpoints so separate repositories have to be registered
-                    // with those endpoints in order for the NuGet APIs to search across both in the case where name includes '*'
+                    PSRepositoryInfo currentRepository = repositoriesToSearch[i];
 
-                    // detect if Script repository needs to be added and/or Module repository needs to be skipped
-                    Uri poshTestGalleryScriptsUri = new Uri("https://www.poshtestgallery.com/api/v2/items/psscript/");
-                    PSRepositoryInfo poshTestGalleryScripts = new PSRepositoryInfo(_poshTestGalleryScriptsRepoName, poshTestGalleryScriptsUri, repositoriesToSearch[i].Priority, trusted: false, credentialInfo: null);
-                    if (_type == ResourceType.None)
+                    // Explicitly passed in Credential takes precedence over repository CredentialInfo.
+                    if (_networkCredential == null && currentRepository.CredentialInfo != null)
                     {
-                        _cmdletPassedIn.WriteVerbose("Null Type provided, so add PoshTestGalleryScripts repository");
-                        repositoriesToSearch.Insert(i + 1, poshTestGalleryScripts);
+                        PSCredential repoCredential = Utils.GetRepositoryCredentialFromSecretManagement(
+                            currentRepository.Name,
+                            currentRepository.CredentialInfo,
+                            _cmdletPassedIn);
+
+                        var username = repoCredential.UserName;
+                        var password = repoCredential.Password;
+
+                        _networkCredential = new NetworkCredential(username, password);
```

Comment:
Declaring variables above is not necessary.

```suggestion
                        _networkCredential = new NetworkCredential(
                            repoCredential.UserName,
                            repoCredential.Password);
```